### PR TITLE
deprecates cloud api

### LINKED ISSUE #1111: deprecate `ploomber cloud`
all the commands that begin with `cloud_` will be deprecated soon: https://github.com/ploomber/ploomber/blob/b69a6de7fc513fe27aababdae3d62cf350e0e04c/src/ploomber_cli/cli.py#L438

so we need to display a FutureWarning with the following message: `ploomber cloud will be deprecated and replaced with a new system. If you need help migrating, send us a message to: https://ploomber.io/community`

we also need to add a note (like the one we have [here](https://docs.ploomber.io/en/latest/user-guide/cli.html)) to all sections under [cloud](https://docs.ploomber.io/en/latest/cloud/api-key.html) with the following message: `Ploomber Cloud will be deprecated and replaced with a new system. If you need help migrating, send us a message to: https://ploomber.io/community`

### PR DESCRIPTION
**Pull request by muddi900**
## Describe your changes

Added the warning and note for cloud deprecation

## Issue number

Closes #1111

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

I am having trouble building the docs. Otherwise seems to be fine.

----
:books: Documentation preview :books:: https://ploomber--1113.org.readthedocs.build/en/1113/

### CODE REVIEW

File: doc/user-guide/cli.rst [hunk 1]
```diff
@@ -3,6 +3,8 @@ Command-line interface
 
 .. note::  This is an introductory tutorial to the command line interface; for a complete API description, see: :doc:`../api/cli`.
 
+.. note::  Ploomber Cloud will be deprecated and replaced with a new system. If you need help migrating, send us a message to: https://ploomber.io/community
```
Comment (edublancas) [inline doc/user-guide/cli.rst:6]: please use the rst syntax to add links

File: src/ploomber_cli/cli.py [hunk 2]
```diff
@@ -13,6 +14,22 @@
 VERSION_KWARGS = dict(package_name="ploomber") if CLICK_VERSION >= 8 else dict()
 
 
+def cloud_deprecation_warning():
+    """A resuable function to pass a warning to all cloud_ functions
+
+
+    Return: None
+    """
```
Comment (edublancas) [inline src/ploomber_cli/cli.py:22]: no need to add a docstring here

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/ploomber_cli/cli.py (hunk 2 above)
Comment (muddi900) [reply src/ploomber_cli/cli.py:22]: The lint fails otherwise. Should I remove it anyway?

Comment (edublancas) [reply src/ploomber_cli/cli.py:22]: I don't think the lint requires docstrings. what error did you get?

Comment (muddi900) [reply src/ploomber_cli/cli.py:22]: Seems to have passed this time. Squashing and pushing in a moment.

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
great!

**Pull request merged**